### ISSUE TITLE
Devices stopped working all of a sudden

### ISSUE THREAD

**Issue by orevron** [itavero/homebridge-z2m#370]
**Describe the bug**
i have bunch of devices connected to this plugin and they all worked perfect, but for some reason after a restart to the zigbee2mqtt server the devices stopped working on HomeKit app.
I can still see them in the zigbee2mqtt server and and also when running homebridge in debug mode i can see the devices are connected to homebridge and logging healthy logs like
```
[26/01/2022, 00:48:44] [zigbee2mqtt] Handled device update for tom_button: {"battery":100,"click":"many","linkquality":90,"voltage":3012}
```
but nothing is happening in the HomeKit app.
wonder if i miss something with configuration of something like that, because everything worked perfect minutes ago.

**Related devices**
all my devices stopped working suddenly 

**To Reproduce**
Not sure how to reproduce the issue but it happened after a restart to the zigbee2mqtt server

**Expected behavior**
I see the devices in the HomeKit app and also see the incoming messages in the homebridge logs but nothing is getting triggered in the app.

**MQTT messages**
zigbee2mqtt message
```
info  2022-01-26 00:56:29: MQTT publish: topic 'home/office_button', payload '{"battery":90,"click":"single","linkquality":159,"voltage":2982}'
```
homebridge message
```
[26/01/2022, 00:57:33] [zigbee2mqtt] Handled device update for office_button: {"battery":90,"click":"single","linkquality":156,"voltage":2982}
```

**Versions used**
Please provide the version of the following pieces of software:
 - This plugin 1.6.2
 - Homebridge 1.4.0
 - Zigbee2MQTT 1.22.0
 - Homebridge Config UI X 4.42.0

please any help would be appreciated!

**Comment by itavero** [comment]
Are *all* devices not working? If so, want kind of devices so you used?

With regards to the information you've shared: it contains a `click` property, which is probably a legacy value and **not** part of the `exposes` information provided by Zigbee2MQTT. Therefor I don't think it will be handled by the plugin.

**Comment by itavero** [comment]
Also, please share the Homebridge logs from around the time that Zigbee2MQTT restarted. It should show that Zigbee2MQTT goes offline and might show some devices being recreated.

**Comment by orevron** [comment]
thank you for your help again! very appreciated!
this is the devices i used to work with:
![image](https://user-images.githubusercontent.com/20145882/151116991-1d1ddee2-5119-4f0a-a2ab-c1a07109c68a.png)

not sure i understood the thing with the legacy click but things used too work i mean that click used to do something in the homekit app...

when i restart the zigbee2mqtt server the homebridge logs shows that the server has gone offline but it never gets connected again, i need to restart the homebridge server to get them connected again.
and this is the homebridge logs after a restart to homebridge to see it getting connected to the z2m server:
```
[26/01/2022, 09:04:59] [Homebridge UI] Homebridge Config UI X v4.42.0 is listening on :: port 8581
[26/01/2022, 09:05:00] [HB Supervisor] Starting Homebridge with extra flags: -I
[26/01/2022, 09:05:00] [HB Supervisor] Started Homebridge v1.4.0 with PID: 3332
[26/01/2022, 09:05:02] Loaded config.json with 3 accessories and 10 platforms.
[26/01/2022, 09:05:02] Loaded 38 cached accessories from cachedAccessories.
[26/01/2022, 09:05:03] ---
[26/01/2022, 09:05:05] Loaded plugin: homebridge-config-ui-x@4.42.0
[26/01/2022, 09:05:05] Registering platform 'homebridge-config-ui-x.config'
[26/01/2022, 09:05:05] ---
[26/01/2022, 09:05:05] Loaded plugin: homebridge-dummy@0.5.0
[26/01/2022, 09:05:05] Registering accessory 'homebridge-dummy.DummySwitch'
[26/01/2022, 09:05:05] ---
[26/01/2022, 09:05:05] Loaded plugin: homebridge-ewelink@8.11.2
[26/01/2022, 09:05:05] Registering platform 'homebridge-ewelink.eWeLink'
[26/01/2022, 09:05:05] ---
[26/01/2022, 09:05:05] Loaded plugin: homebridge-mi-aqara@0.8.1
[26/01/2022, 09:05:05] Registering platform 'homebridge-mi-aqara.MiAqaraPlatform'
[26/01/2022, 09:05:05] ---
[26/01/2022, 09:05:06] Loaded plugin: homebridge-samsung-tizen@5.1.0
[26/01/2022, 09:05:06] Registering platform 'homebridge-samsung-tizen.SamsungTizen'
[26/01/2022, 09:05:06] ---
[26/01/2022, 09:05:06] Loaded plugin: homebridge-sensibo-ac@2.2.0
[26/01/2022, 09:05:06] Registering platform 'homebridge-sensibo-ac.SensiboAC'
[26/01/2022, 09:05:06] ---
[26/01/2022, 09:05:06] Loaded plugin: homebridge-switcher-platform@4.1.0
[26/01/2022, 09:05:06] Registering platform 'homebridge-switcher-platform.SwitcherPlatform'
[26/01/2022, 09:05:06] ---
[26/01/2022, 09:05:07] Loaded plugin: homebridge-unifi-protect@5.0.6
[26/01/2022, 09:05:07] Registering platform 'homebridge-unifi-protect.UniFi Protect'
[26/01/2022, 09:05:07] ---
[26/01/2022, 09:05:08] Loaded plugin: homebridge-xiaomi-fan@1.5.7
[26/01/2022, 09:05:08] Registering platform 'homebridge-xiaomi-fan.xiaomifan'
[26/01/2022, 09:05:08] ---
[26/01/2022, 09:05:09] Loaded plugin: homebridge-xiaomi-roborock-vacuum@0.20.1
[26/01/2022, 09:05:09] Registering accessory 'homebridge-xiaomi-roborock-vacuum.XiaomiRoborockVacuum'
[26/01/2022, 09:05:09] ---
[26/01/2022, 09:05:09] Loaded plugin: homebridge-yeelight-wifi@2.3.3
[26/01/2022, 09:05:09] Registering platform 'homebridge-yeelight-wifi.yeelight'
[26/01/2022, 09:05:09] ---
[26/01/2022, 09:05:09] Loaded plugin: homebridge-z2m@1.6.2
[26/01/2022, 09:05:09] Registering platform 'homebridge-z2m.zigbee2mqtt'
[26/01/2022, 09:05:09] ---
[26/01/2022, 09:05:09] Loading 10 platforms...
[26/01/2022, 09:05:09] [SamsungTizen] Initializing SamsungTizen platform...
[26/01/2022, 09:05:09] [Yeelight] Initializing yeelight platform...
[26/01/2022, 09:05:09] [MiAqaraPlatform] Initializing MiAqaraPlatform platform...
[26/01/2022, 09:05:09] [MiAqaraPlatform] [INFO]**************************************************************
[26/01/2022, 09:05:09] [MiAqaraPlatform] [INFO]           MiAqaraPlatform v0.8.1 By YinHang
[26/01/2022, 09:05:09] [MiAqaraPlatform] [INFO]  GitHub: https://github.com/YinHangCode/homebridge-mi-aqara
[26/01/2022, 09:05:09] [MiAqaraPlatform] [INFO]                                         QQ Group: 107927710
[26/01/2022, 09:05:09] [MiAqaraPlatform] [INFO]**************************************************************
[26/01/2022, 09:05:09] [MiAqaraPlatform] [INFO]start success...
[26/01/2022, 09:05:09] [MiAqaraPlatform] [INFO]config gateways: f0b429cc8504
[26/01/2022, 09:05:09] [MiAqaraPlatform] [INFO]binding to the default interface
[26/01/2022, 09:05:09] [Switcher Platform] Initializing SwitcherPlatform platform...
[26/01/2022, 09:05:09] [xiaomifan] Initializing xiaomifan platform...
[26/01/2022, 09:05:09] [SensiboAC] Initializing SensiboAC platform...
[26/01/2022, 09:05:09] [eWeLink] Initializing eWeLink platform...
[26/01/2022, 09:05:09] [eWeLink] Initialising plugin v8.11.2 | Node v16.13.2 | HB v1.4.0 | HAPNodeJS v0.10.0...
[26/01/2022, 09:05:09] [Config] Initializing config platform...
[26/01/2022, 09:05:09] [Config] Running in Service Mode
[26/01/2022, 09:05:09] [UniFi Protect] Initializing UniFi Protect platform...
[26/01/2022, 09:05:09] [zigbee2mqtt] Initializing zigbee2mqtt platform...
[26/01/2022, 09:05:09] [zigbee2mqtt] Connecting to MQTT server at mqtt://192.168.1.221:1883
[26/01/2022, 09:05:10] Loading 3 accessories...
[26/01/2022, 09:05:10] [Vaccum Robot] Initializing XiaomiRoborockVacuum accessory...
[26/01/2022, 09:05:10] [Vaccum Robot] INF createRoom | undefined | Room Living Room (0)
[26/01/2022, 09:05:10] [Vaccum Robot] INF createRoom | undefined | Room Kitchen (1)
[26/01/2022, 09:05:10] [Vaccum Robot] INF createRoom | undefined | Room Bathroom (2)
[26/01/2022, 09:05:10] [Vaccum Robot] INF createRoom | undefined | Room Office (3)
[26/01/2022, 09:05:10] [Vaccum Robot] INF createRoom | undefined | Room Bedroom (4)
[26/01/2022, 09:05:10] [Bedroom Automation] Initializing DummySwitch accessory...
[26/01/2022, 09:05:10] [Bedroom Automation] Setting switch to false
[26/01/2022, 09:05:10] [Office Automation] Initializing DummySwitch accessory...
[26/01/2022, 09:05:10] [Office Automation] Setting switch to true
[26/01/2022, 09:05:10] [Yeelight] Loaded accessory color-fca202.
[26/01/2022, 09:05:10] [Yeelight] Loaded accessory strip6-f28e21.
[26/01/2022, 09:05:10] [Yeelight] Loaded accessory colora-830d56.
[26/01/2022, 09:05:10] [Yeelight] Loaded accessory colora-831467.
[26/01/2022, 09:05:10] [zigbee2mqtt] Restoring accessory: office_button (0x00158d0002c48270)
[26/01/2022, 09:05:10] [zigbee2mqtt] Mapping of property 'action' of device 'office_button':
Button 1:
	- SINGLE: single
	- DOUBLE: double
	- LONG  : hold
Button 2 #ext1:
	- SINGLE: triple
	- DOUBLE: quadruple
Button 3 (many):
	- SINGLE: many
[26/01/2022, 09:05:10] [zigbee2mqtt] Restoring accessory: office_door (0x00158d000588ad0d)
[26/01/2022, 09:05:10] [zigbee2mqtt] Restoring accessory: network_temperature (0x00158d00042d5543)
[26/01/2022, 09:05:10] [zigbee2mqtt] Restoring accessory: office_vibration (0x00158d0006370b01)
[26/01/2022, 09:05:10] [zigbee2mqtt] Mapping of property 'action' of device 'office_vibration':
Button 1 (drop):
	- SINGLE: drop
Button 2 (tilt):
	- SINGLE: tilt
Button 3 (vibration):
	- SINGLE: vibration
[26/01/2022, 09:05:10] [zigbee2mqtt] Restoring accessory: office_motion (0x00158d00027ce768)
[26/01/2022, 09:05:10] [zigbee2mqtt] Restoring accessory: entrance_switch (0x00158d0005c10071)
[26/01/2022, 09:05:10] [zigbee2mqtt] Mapping of property 'action' of device 'entrance_switch':
Button 1 (both):
	- SINGLE: single_both
	- DOUBLE: double_both
	- LONG  : hold_both
Button 2 (left):
	- SINGLE: single_left
	- DOUBLE: double_left
	- LONG  : hold_left
Button 3 (right):
	- SINGLE: single_right
	- DOUBLE: double_right
	- LONG  : hold_right
[26/01/2022, 09:05:11] [zigbee2mqtt] Restoring accessory: tom_button (0x00158d00035563b1)
[26/01/2022, 09:05:11] [zigbee2mqtt] Mapping of property 'action' of device 'tom_button':
Button 1:
	- SINGLE: single
	- DOUBLE: double
	- LONG  : hold
Button 2 #ext1:
	- SINGLE: triple
	- DOUBLE: quadruple
Button 3 (many):
	- SINGLE: many
[26/01/2022, 09:05:11] [Yeelight] Searching for known devices...
[26/01/2022, 09:05:11] [Yeelight] Sending search request...
[26/01/2022, 09:05:11] [xiaomifan] Init - initializing devices
[26/01/2022, 09:05:11] [xiaomifan] [Xiaomi Fan 2] Init - got fan configuration, initializing device with name: Xiaomi Fan 2
[26/01/2022, 09:05:11] [eWeLink] Plugin initialised. Setting up accessories....
Setup Payload:
X-HM://0024THI6BQCSS
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:

    ┌────────────┐
    │ 815-52-195 │
    └────────────┘

[26/01/2022, 09:05:11] [MiAqaraPlatform] [INFO]Aqara LAN protocol server is listening on port: 9898
[26/01/2022, 09:05:11] Homebridge v1.4.0 (HAP v0.10.0) (Homebridge 35DB) is running on port 51506.
[26/01/2022, 09:05:11] [Yeelight] Received advertisement from fca202.
[26/01/2022, 09:05:11] [Yeelight] Device color-fca202 supports brightness
[26/01/2022, 09:05:11] [Yeelight] Device color-fca202 supports color
[26/01/2022, 09:05:11] [Yeelight] Device color-fca202 supports color temperature
[26/01/2022, 09:05:11] [homebridge-yeelight-wifi] This plugin generated a warning from the characteristic 'On': characteristic value expected boolean and received undefined. See https://git.io/JtMGR for more info.
[26/01/2022, 09:05:11] [Yeelight] Initialized device  (192.168.1.163:55443).
[26/01/2022, 09:05:11] [homebridge-yeelight-wifi] This plugin generated a warning from the characteristic 'Color Temperature': characteristic was supplied illegal value: number 152.84615384615384 exceeded minimum of 154. See https://git.io/JtMGR for more info.
[26/01/2022, 09:05:11] [Yeelight] Received advertisement from fca202.
[26/01/2022, 09:05:11] [Yeelight] Received advertisement from f28e21.
[26/01/2022, 09:05:11] [Yeelight] Device strip6-f28e21 supports brightness
[26/01/2022, 09:05:11] [Yeelight] Device strip6-f28e21 supports color
[26/01/2022, 09:05:11] [Yeelight] Device strip6-f28e21 supports color temperature
[26/01/2022, 09:05:11] [homebridge-yeelight-wifi] This plugin generated a warning from the characteristic 'On': characteristic value expected boolean and received undefined. See https://git.io/JtMGR for more info.
[26/01/2022, 09:05:11] [Yeelight] Initialized device  (192.168.1.92:55443).
[26/01/2022, 09:05:11] [Yeelight] Received advertisement from f28e21.
[26/01/2022, 09:05:11] [eWeLink] HTTP/WS clients disabled as eWeLink credentials not configured.
[26/01/2022, 09:05:11] [eWeLink] Obtaining device list from storage.
[26/01/2022, 09:05:11] [zigbee2mqtt] Connected to MQTT server
[26/01/2022, 09:05:12] [zigbee2mqtt] Using Zigbee2MQTT v1.22.0 (identified via home/bridge/info)
```

**Comment by itavero** [comment]
The connection is with the MQTT server, not Zigbee2MQTT. It doesn't loose the connection to the MQTT server, so that's why it does not reconnect (it stays connected).

If you have enabled the `legacy` option for certain devices, I think that is probably causing it to send data not in the `exposes` information provided by Zigbee2MQTT.

I think the door sensor doesn't have a "legacy" option (at least it's not listed on the Zigbee2MQTT website). Have you tried triggering that to see if the status in the Home app is updated?

**Comment by orevron** [comment]
weird because when i restarting the z2m server i see this log in the homebridge logs
```
[26/01/2022, 10:26:06] [zigbee2mqtt] Zigbee2MQTT is OFFLINE!
```
but it never comes back

anyway this is the mosquitto logs when i restart z2m server:
```
1643180480: New client connected from 127.0.0.1:54936 as mqttjs_5ef15bf4 (p2, c1, k60, u'orevron').
1643180684: Client mqttjs_09076862 closed its connection.
1643180711: New connection from 192.168.1.140:46294 on port 1883.
1643180711: New client connected from 192.168.1.140:46294 as mqttjs_0ed096cb (p2, c1, k60, u'orevron').
1643181758: Saving in-memory database to /var/lib/mosquitto//mosquitto.db.
1643183559: Saving in-memory database to /var/lib/mosquitto//mosquitto.db.
1643185360: Saving in-memory database to /var/lib/mosquitto//mosquitto.db.
1643185566: Client mqttjs_5ef15bf4 disconnected.
1643185587: New connection from 127.0.0.1:54938 on port 1883.
1643185587: New client connected from 127.0.0.1:54938 as mqttjs_fd784397 (p2, c1, k60, u'orevron').
1643186189: Client mqttjs_fd784397 disconnected.
1643186209: New connection from 127.0.0.1:54944 on port 1883.
1643186209: New client connected from 127.0.0.1:54944 as mqttjs_d0086a7b (p2, c1, k60, u'orevron').
```

and this is the z2m logs:
```
warn  2022-01-26 10:36:50: Using experimental new availability feature
info  2022-01-26 10:36:50: Started frontend on port 0.0.0.0:8080
info  2022-01-26 10:36:50: MQTT publish: topic 'home/bridge/config', payload '{"commit":"df0543d4","coordinator":{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20210319,"transportrev":2},"type":"zStack3x0"},"log_level":"info","network":{"channel":11,"extendedPanID":"0x00124b0018e32e5c","panID":6754},"permit_join":false,"version":"1.22.0"}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/office_button', payload '{"battery":90,"voltage":2982}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/office_door', payload '{"battery":100,"contact":false,"voltage":3015}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/network_temperature', payload '{"battery":100,"humidity":39.67,"pressure":1009,"temperature":17.72,"voltage":3005}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/office_vibration', payload '{"angle":5,"angle_x":3,"angle_x_absolute":87,"angle_y":1,"angle_y_absolute":89,"angle_z":87,"battery":86,"strength":6,"voltage":2975}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/office_motion', payload '{"battery":100,"occupancy":false,"voltage":3015}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/entrance_switch', payload '{"battery":97,"voltage":2995}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/tom_button', payload '{"battery":100,"voltage":3012}'
info  2022-01-26 10:39:33: MQTT publish: topic 'home/office_button', payload '{"battery":84,"linkquality":156,"voltage":2972}'
info  2022-01-26 10:39:46: MQTT publish: topic 'home/office_vibration', payload '{"angle":5,"angle_x":3,"angle_x_absolute":87,"angle_y":1,"angle_y_absolute":89,"angle_z":87,"battery":86,"linkquality":129,"strength":6,"voltage":2975}'
```

**Comment by orevron** [comment]
weird because when i restarting the z2m server i see this log in the homebridge logs
```
[26/01/2022, 10:26:06] [zigbee2mqtt] Zigbee2MQTT is OFFLINE!
```
but it never comes back

anyway this is the mosquitto logs when i restart z2m server:
```
1643180480: New client connected from 127.0.0.1:54936 as mqttjs_5ef15bf4 (p2, c1, k60, u'orevron').
1643180684: Client mqttjs_09076862 closed its connection.
1643180711: New connection from 192.168.1.140:46294 on port 1883.
1643180711: New client connected from 192.168.1.140:46294 as mqttjs_0ed096cb (p2, c1, k60, u'orevron').
1643181758: Saving in-memory database to /var/lib/mosquitto//mosquitto.db.
1643183559: Saving in-memory database to /var/lib/mosquitto//mosquitto.db.
1643185360: Saving in-memory database to /var/lib/mosquitto//mosquitto.db.
1643185566: Client mqttjs_5ef15bf4 disconnected.
1643185587: New connection from 127.0.0.1:54938 on port 1883.
1643185587: New client connected from 127.0.0.1:54938 as mqttjs_fd784397 (p2, c1, k60, u'orevron').
1643186189: Client mqttjs_fd784397 disconnected.
1643186209: New connection from 127.0.0.1:54944 on port 1883.
1643186209: New client connected from 127.0.0.1:54944 as mqttjs_d0086a7b (p2, c1, k60, u'orevron').
```

and this is the z2m logs:
```
warn  2022-01-26 10:36:50: Using experimental new availability feature
info  2022-01-26 10:36:50: Started frontend on port 0.0.0.0:8080
info  2022-01-26 10:36:50: MQTT publish: topic 'home/bridge/config', payload '{"commit":"df0543d4","coordinator":{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20210319,"transportrev":2},"type":"zStack3x0"},"log_level":"info","network":{"channel":11,"extendedPanID":"0x00124b0018e32e5c","panID":6754},"permit_join":false,"version":"1.22.0"}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/office_button', payload '{"battery":90,"voltage":2982}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/office_door', payload '{"battery":100,"contact":false,"voltage":3015}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/network_temperature', payload '{"battery":100,"humidity":39.67,"pressure":1009,"temperature":17.72,"voltage":3005}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/office_vibration', payload '{"angle":5,"angle_x":3,"angle_x_absolute":87,"angle_y":1,"angle_y_absolute":89,"angle_z":87,"battery":86,"strength":6,"voltage":2975}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/office_motion', payload '{"battery":100,"occupancy":false,"voltage":3015}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/entrance_switch', payload '{"battery":97,"voltage":2995}'
info  2022-01-26 10:36:50: MQTT publish: topic 'home/tom_button', payload '{"battery":100,"voltage":3012}'
info  2022-01-26 10:39:33: MQTT publish: topic 'home/office_button', payload '{"battery":84,"linkquality":156,"voltage":2972}'
info  2022-01-26 10:39:46: MQTT publish: topic 'home/office_vibration', payload '{"angle":5,"angle_x":3,"angle_x_absolute":87,"angle_y":1,"angle_y_absolute":89,"angle_z":87,"battery":86,"linkquality":129,"strength":6,"voltage":2975}'
```

**Comment by orevron** [comment]
I figured out what caused the issue, but not how it could it possible. 
I added 4 new HomePods to my network. 
One of them took over and acted as the Apple bridge and apparently something was not good with it's configuration because when I opened it's configuration the app said that i can't manage this homepod when we're not on the same network (which we were clearly in the same network).
I disconnected this homepod and let other homepod in the house take the bridge control and everything came back to life again.

Thanks a lot again for trying to help and for this awesome plugin!
